lib/vector-core/buffers/src/lib.rs

```diff
@@ -178,7 +193,7 @@ impl<T, S: Sink<T> + Unpin> Sink<T> for DropWhenFull<S> {
                 *this.drop = true;
                 Poll::Ready(Ok(()))
             }
-            error => error,
+            error @ std::task::Poll::Ready(..) => error,
```

Comment:
Is this to be more explicit that we're doing passthrough in the error case?

Reply:
Yep, this is a clippy pedantic ding, specifically https://rust-lang.github.io/rust-clippy/master/#wildcard_enum_match_arm. We'd flagged it off in the top-level vector-core but here, at least, I buy being explicit.